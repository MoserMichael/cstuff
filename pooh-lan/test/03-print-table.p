

days =  {
          'sun'  : 'Sunday',
          'mon'  : 'Monday' ,
          'tue'  : 'Tuesday' ,
          'wed'  : 'Wednesday' ,
          'thur' : 'Thursday' ,
          'fri'  : 'Friday' ,
          'sat'  : 'Shabbat'
        }

daykeys = keys( ~table days )
i = 1

println( ~msg '**** value { keys  } ***' )

while daykeys[ i ] != Nill
    println( ~msg  days{ daykeys[ i ] }   )
    i = i + 1
end

println( ~msg '*** value ***' )

daykeys = values( ~table days )
i = 1

while daykeys[ i ] != Nill
    println( ~msg  daykeys[ i ]    )
    i = i + 1
end

println( ~msg '*** each ***' )
v = each(  ~table days )

i = 1
while i <= size( ~arg v )
  e = v[ i ] 
  println( ~msg e[ 1 ] )
  println( ~msg e[ 2 ] )
  i = i + 1
end
