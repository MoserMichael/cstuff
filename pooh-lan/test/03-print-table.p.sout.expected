**** value { keys  } ***
Monday
Friday
Tuesday
Thursday
Shabbat
Sunday
Wednesday
*** value ***
Monday
Friday
Tuesday
Thursday
Shabbat
Sunday
Wednesday
*** each ***
mon
Monday
fri
Friday
tue
Tuesday
thur
Thursday
sat
Shabbat
sun
Sunday
wed
Wednesday
